test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -133,6 +133,56 @@ test.describe(
 				)
 			).toBeVisible();
 		} );
+
+		test( 'default to my most recently created menu', async ( {
+			admin,
+			page,
+			editor,
+			requestUtils,
+		} ) => {
+			await admin.createNewPost();
+			await requestUtils.createNavigationMenu( {
+				title: 'Test Menu 1',
+				content:
+					'<!-- wp:navigation-link {"label":"WordPress","type":"custom","url":"http://www.wordpress.org/","kind":"custom","isTopLevelLink":true} /-->',
+			} );
+
+			// The two menus are created with the same timestamp, so we need to wait a bit to ensure the second one is created after the first one.
+			await editor.page.waitForTimeout( 2000 );
```

Comment:
In my environment both menus were being created with the same timestamp (same second) and when being evaluated to check for the newest one, the older one was the one being returned.

Reply:
Could you elaborate a bit more on why this is needed? Is this a limitation of the REST API?

Reply:
I was looking for the code to show you why I did this and it turns out [yesterday it was refactored](https://github.com/WordPress/gutenberg/pull/48916/files) and this is not needed anymore, so I removed it!

Reply:
It looks like the function that decides which menu to show has [this comment](https://github.com/WordPress/gutenberg/blob/621c2556e843c7c2c0210e13115d29026f6854c4/packages/block-library/src/navigation/edit/index.js#L204). It seems like the API is having trouble deciding which one was the more recent if both are created during the same second? What do you think we should do?

Reply:
If it's a legit bug of the API then we should open an issue about it? In the meantime, we can use `page.waitForTimeout( 1000 )` as a temporary hack, but make sure to add some `FIXME` comments above it. Thanks for looking into it! ❤️